share/plugins/colornotes.qml

```diff
@@ -116,15 +71,17 @@ MuseScore {
                         note.accidental.color = black;
                   }
 
-            for (var i = 0; i < note.dots.length; i++) {
-                  if (note.dots[i]) {
-                        if (note.dots[i].color == black)
-                              note.dots[i].color = colors[note.pitch % 12];
-                        else
-                              note.dots[i].color = black;
+            if (note.dots) {
```

Comment:
At one point my plugin crashed here because I accidentally passed in something that wasn't a note, so figured it was worth keeping as a safety check